### PR TITLE
Feanil/optional datadog

### PR DESCRIPTION
**Pull request by feanil**
@jzoldak @benpatterson 

I moved around the datadog dependency because I don't need it for the admin jenkins we use in devops and so I wanted it to be optional.  I also updated the jenkins_master.yml play you're using to indicate that it is being used by testeng.

### CODE REVIEW

File: playbooks/edx-east/jenkins_testeng.yml [hunk 1]
```diff
@@ -8,6 +8,8 @@
   gather_facts: True
   vars:
     COMMON_DATA_DIR: "/mnt"
+    COMMON_ENABLE_DATADOG: True
   roles:
     - common
+    - datadog
```
Comment (jarv) [inline playbooks/edx-east/jenkins_testeng.yml]: The only place this `COMMON_ENABLE_DATADOG` is used is iat the play level so I think you would want to add a `when: COMMON_ENABLE_DATADOG` condition to the role include

Comment (feanil) [inline playbooks/edx-east/jenkins_testeng.yml]: As a part of this change, I'm now using it within the `jenkins_master/defaults/main.yml`

File: playbooks/edx-east/jenkins_testeng.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Configure a Jenkins master instance
+# Configure a Jenkins instance for testeng.
```
Comment (jzoldak) [inline playbooks/edx-east/jenkins_testeng.yml]: This is still the testeng master, right? If so, let's keep the word master in there too.

**Comment by feanil** [comment]
@jzoldak any other comments?  good to merge?

**Comment by jzoldak** [comment]
sure, can you pls change the comment in playbooks/edx-east/jenkins_testeng_master.yml to "Configure a Jenkins master instance for testeng", then :thumbsup:

**Pull request merged**